### PR TITLE
refactor(sh-admin): updated invite flow for the admin dashboard

### PR DESCRIPTION
**Pull request by JoelJacobStephen**
### Ticket
Closes HFE-540

### Description
This PR focuses on improving the current invite flow in the admin dashboard.

### Changes Introduced
- New Success Invite Modal introduced when an invite is successfully made.
- Users can copy the invite link from the newly introduced modal.
- If SMTP is enabled, an email is sent to the user and the admin can copy the invite link
- If SMTP is disabled, the admin can copy the invite link

### Objectives
- [x] Admins can performs a single invite, regardless of the SMTP status
- [x] Change the copy `Send Invite` to `Add user` 
- [x] Remove the `Copy Link` button
- [x] When the user enters an email and clicks on add user, show a success modal with different content depending on the cases below:
- **Case 1:** If SMTP is enabled, an email is sent and admins can copy the invite link. This is mentioned in the modal.
- **Case 2:** If SMTP is disabled, admins can only copy the invite link 	

### Screenshot
image

### Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

**Review by nivedin — CHANGES_REQUESTED** [review]
why do we need a `Copy Link` button in the users/invited page ?

image

also its visible in empty state as well
image

File: packages/hoppscotch-sh-admin/src/pages/users/index.vue [hunk 1]
```diff
@@ -203,6 +203,43 @@
...
+          <div class="bg-primaryDark rounded-r-sm">
+            <UiAutoResetIcon
+              :title="t('users.copy_link')"
+              :icon="{ default: IconCopy, temporary: IconCheck }"
+              @click="copyInviteLink"
+            />
+          </div>
+        </div>
+      </template>
+    </HoppSmartModal>
```
Comment (nivedin) [inline packages/hoppscotch-sh-admin/src/pages/users/index.vue:241]: we can convert this to a modal component

**Review by nivedin — CHANGES_REQUESTED** [review]

**Comment by JoelJacobStephen** [comment]
> why do we need a `Copy Link` button in the users/invited page ?
> 
> image
> 
> also its visible in empty state as well image

@nivedin The link for all the invites shown is the same which is the `baseURL`. That is why there is a signle copy link button. I have now removed copy link button when there are no pending invites.

**Comment by nivedin** [comment]
> > why do we need a `Copy Link` button in the users/invited page ?
> > image
> > also its visible in empty state as well image
> 
> @nivedin The link for all the invites shown is the same which is the `baseURL`. That is why there is a signle copy link button. I have now removed copy link button when there are no pending invites.

okay gotcha, can we maybe change the text to `Copy Invite Link` or something, its difficult to understand the context wdyt @liyasthomas

**Comment by JoelJacobStephen** [comment]
@nivedin I've changed the button text to `Copy Invite Link` after taking approval from @liyasthomas. I have pushed the changes.

**Review by nivedin — APPROVED** [review]

**Pull request merged**